core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1504,60 +1504,9 @@ class KafkaApis(val requestChannel: RequestChannel,
       isAcknowledgeDataPresent
     }
 
-    val authorizedTopics = authHelper.filterByAuthorized(
-      request.context,
-      READ,
-      TOPIC,
-      shareFetchData.asScala
-    )(_._1.topicPartition.topic)
-
     var shareAcknowledgeResponse : ShareAcknowledgeResponse = null
     var shareFetchResponse : ShareFetchResponse = null
 
-    // Handling the Acknowledgements from the ShareFetchRequest
-    if (!authHelper.authorize(request.context, READ, GROUP, groupId)) {
-      shareAcknowledgeResponse = shareFetchRequest.getErrorAcknowledgeResponse(AbstractResponse.DEFAULT_THROTTLE_TIME, Errors.GROUP_AUTHORIZATION_FAILED.exception) match {
-        case response: ShareAcknowledgeResponse => response
-        case _ => null
-      }
-    } else {
-      val isAcknowledgeDataPresent = isAcknowledgeDataPresentInFetchRequest()
-      if (shareSessionEpoch == ShareFetchMetadata.INITIAL_EPOCH) {
-        if (isAcknowledgeDataPresent) {
-          // If this is a full fetch request, it should not contain any acknowledgement Data
-          shareAcknowledgeResponse = shareFetchRequest.getErrorAcknowledgeResponse(AbstractResponse.DEFAULT_THROTTLE_TIME, Errors.INVALID_REQUEST.exception) match {
-            case response: ShareAcknowledgeResponse => response
-            case _ => null
-          }
-        }
-        else {
-          shareAcknowledgeResponse = ShareAcknowledgeResponse.of(
-            Errors.NONE,
-            AbstractResponse.DEFAULT_THROTTLE_TIME,
-            new util.LinkedHashMap[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData],
-            Collections.emptyList())
-        }
-      }
-      else {
-        if (!isAcknowledgeDataPresent) {
-          shareAcknowledgeResponse = ShareAcknowledgeResponse.of(
-            Errors.NONE,
-            AbstractResponse.DEFAULT_THROTTLE_TIME,
-            new util.LinkedHashMap[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData],
-            Collections.emptyList())
-        }
-        val shareSessionError: Errors = sharePartitionManager.acknowledgeShareSessionCacheUpdate(groupId, Uuid.fromString(memberId), shareSessionEpoch, true)
```

Comment:
Hi, thanks for the PR. I see the use of `sharePartitionManager.acknowledgeShareSessionCacheUpdate` has been removed. If that's the case, then wouldn't the acknowledgements proceed successfully even when the share session id/epoch sent by the consumer is wrong ?